haystack/models.py

```diff
@@ -95,7 +95,13 @@ def _set_object(self, obj):
 
     def _get_model(self):
         if self._model is None:
-            self._model = models.get_model(self.app_label, self.model_name)
+            try:
+                self._model = models.get_model(self.app_label, self.model_name)
+            except LookupError:
+                # this changed in change 1.7 to throw an error instead of
+                # returning None when the model isn't found. So catch the
+                # lookup error and keep self._model == None.
+                pass
```

Comment:
```
get_model raises LookupError instead of returning None when no model is found.
```

https://docs.djangoproject.com/en/1.7/releases/1.7/#introspecting-applications